lib/galaxy/jobs/__init__.py

```diff
@@ -1621,10 +1621,12 @@ def setup_external_metadata( self, exec_dir=None, tmp_dir=None,
         command = base_command
         if resolve_metadata_dependencies:
             metadata_tool = self.app.toolbox.get_tool("__SET_METADATA__")
-            dependency_shell_commands = metadata_tool.build_dependency_shell_commands(job_directory=self.working_directory)
-            if dependency_shell_commands:
-                dependency_shell_commands = "; ".join(dependency_shell_commands)
-                command = "%s; %s" % (dependency_shell_commands, command)
+            if metadata_tool is not None:
+                # Due to tool shed hacks for migrate and installed tool tests...
```

Comment:
Sure - but I have refactored the hack to its own method as part of #2016 - so I'd like the reference be added to dev instead of 16.01 (so I'd merge this as is - and let https://github.com/galaxyproject/galaxy/pull/2016/commits/9705cea615da20e9a9758407db055143a3afa7da add the comment when #2016 is merged).